bin/lcov-result-merger.js

```diff
@@ -15,11 +15,25 @@ if (process.argv.length < 3) {
   process.exit(1)
 }
 
-var files = process.argv[2]
-var outputFile = process.argv[3]
+var userArgs = process.argv.slice(2)
+
+var firstConfigFlag = userArgs.find(function (value) {
+  return typeof value === 'string' && value.substring(0, 2) === '--'
+})
+var firstFlagIndex = firstConfigFlag ? userArgs.indexOf(firstConfigFlag) : -1
+
+var configFlags = firstFlagIndex > -1
+  ? userArgs.slice(firstFlagIndex).reduce(function (acc, cur) {
+    acc[cur.substring(2)] = true
+    return acc
+  }, {})
+  : {}
+
+var files = userArgs[0]
+var outputFile = firstFlagIndex > 1 ? userArgs[1] : null
```

Comment:
there's something off with parsing the `outputFile`. I quickly copied the code to a file and tested a few things:

```js
var userArgs = process.argv.slice(2)

var firstConfigFlag = userArgs.find(function (value) {
  return typeof value === 'string' && value.substring(0, 2) === '--'
})
var firstFlagIndex = firstConfigFlag ? userArgs.indexOf(firstConfigFlag) : -1

var configFlags = firstFlagIndex > -1
  ? userArgs.slice(firstFlagIndex).reduce(function (acc, cur) {
    acc[cur.substring(2)] = true
    return acc
  }, {})
  : {}

var files = userArgs[0]
var outputFile = firstFlagIndex > 1 ? userArgs[1] : null

console.log(configFlags, files, outputFile)
```

```bash
» node test.js pattern output
{} pattern null
» node test.js pattern output --foo-bar-baz
{ 'foo-bar-baz': true } pattern output
» node test.js pattern output
{} pattern null
» node test.js pattern --foo-bar-baz
{ 'foo-bar-baz': true } pattern null
```

the output is never set if not also the flag is set.

Reply:
Ya, that whole chunk was feeling pretty hacky the entire time I was writing it. Brought in Yargs for the argument string parsing.